pycode/memilio-simulation/memilio/simulation/bindings/models/osecir.cpp

```diff
@@ -48,98 +48,75 @@
 {
 
-//select only the first node of the graph of each run, used for parameterstudy with single nodes
-template <class Sim>
-std::vector<Sim> filter_graph_results(
-    std::vector<mio::Graph<mio::SimulationNode<double, Sim>, mio::MobilityEdge<double>>>&& graph_results)
-{
-    std::vector<Sim> results;
-    results.reserve(graph_results.size());
-    for (auto i = size_t(0); i < graph_results.size(); ++i) {
-        results.emplace_back(std::move(graph_results[i].nodes()[0].property.get_simulation()));
-    }
-    return std::move(results);
-}
-
 /*
  * @brief bind ParameterStudy for any model
  */
-template <class Simulation>
+template <class Sim>
 void bind_ParameterStudy(py::module_& m, std::string const& name)
 {
-    pymio::bind_class<mio::ParameterStudy<double, Simulation>, pymio::EnablePickling::Never>(m, name.c_str())
-        .def(py::init<const typename Simulation::Model&, double, double, size_t>(), py::arg("model"), py::arg("t0"),
-             py::arg("tmax"), py::arg("num_runs"))
-        .def(py::init<const mio::Graph<typename Simulation::Model, mio::MobilityParameters<double>>&, double, double,
-                      double, size_t>(),
-             py::arg("model_graph"), py::arg("t0"), py::arg("tmax"), py::arg("dt"), py::arg("num_runs"))
-        .def_property("num_runs", &mio::ParameterStudy<double, Simulation>::get_num_runs,
-                      &mio::ParameterStudy<double, Simulation>::set_num_runs)
-        .def_property("tmax", &mio::ParameterStudy<double, Simulation>::get_tmax,
-                      &mio::ParameterStudy<double, Simulation>::set_tmax)
-        .def_property("t0", &mio::ParameterStudy<double, Simulation>::get_t0,
-                      &mio::ParameterStudy<double, Simulation>::set_t0)
-        .def_property_readonly("model", py::overload_cast<>(&mio::ParameterStudy<double, Simulation>::get_model),
-                               py::return_value_policy::reference_internal)
-        .def_property_readonly("model",
-                               py::overload_cast<>(&mio::ParameterStudy<double, Simulation>::get_model, py::const_),
+    using GraphT      = mio::Graph<mio::SimulationNode<double, Sim>, mio::MobilityEdge<double>>;
+    using SimulationT = mio::GraphSimulation<double, GraphT, double, double>;
+    using ParametersT = mio::Graph<typename Sim::Model, mio::MobilityParameters<double>>;
+    using StudyT      = mio::ParameterStudy<ParametersT, double>;
+
+    const auto create_simulation = [](const ParametersT& g, double t0, double dt, size_t) {
+        auto copy = g;
+        return mio::make_sampled_graph_simulation<double, Sim>(draw_sample(copy), t0, dt, dt);
+    };
+
+    pymio::bind_class<StudyT, pymio::EnablePickling::Never>(m, name.c_str())
+        .def(py::init<const ParametersT&, double, double, double, size_t>(), py::arg("parameters"), py::arg("t0"),
+             py::arg("tmax"), py::arg("dt"), py::arg("num_runs"))
+        .def_property_readonly("num_runs", &StudyT::get_num_runs)
```

Comment:
Add py::arg() and strings as documentation. Documentation especially for the functions that should be called by the user.